glam/api/migrations/0020_transpose_fog.py

```diff
@@ -0,0 +1,68 @@
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('api', '0019_fog_aggregation'),
+    ]
+
+    operations = [
+        migrations.DeleteModel(
+            name='FOGAggregation',
+        ),
+
+        migrations.CreateModel(
+            name='FOGAggregation',
+            fields=[
+                ('id', models.BigAutoField(primary_key=True, serialize=False)),
+                ('channel', models.CharField(max_length=100)),
+                ('version', models.CharField(max_length=100)),
+                ('ping_type', models.CharField(max_length=100)),
+                ('os', models.CharField(max_length=100)),
+                ('build_id', models.CharField(max_length=100)),
+                ('build_date', models.DateTimeField(null=True)),
+                ('metric', models.CharField(max_length=200)),
+                ('metric_type', models.CharField(max_length=100)),
+                ('metric_key', models.CharField(blank=True, max_length=200)),
+                ('client_agg_type', models.CharField(blank=True, max_length=100)),
+                ('total_users', models.IntegerField()),
+                ('histogram', models.TextField(blank=True, null=True)),
+                ('percentiles', models.TextField(blank=True, null=True)),
+            ],
+            options={
+                'db_table': 'glam_fog_aggregation',
+                'abstract': False,
+            },
+        ),
+        migrations.AddConstraint(
+            model_name='fogaggregation',
+            constraint=models.UniqueConstraint(fields=('channel', 'version', 'ping_type', 'os', 'build_id', 'metric', 'metric_type', 'metric_key', 'client_agg_type'), name='fog_unique_dimensions'),
+        ),
+        migrations.CreateModel(
+            name='FOGAggregationView',
+            fields=[
+                ('id', models.BigAutoField(primary_key=True, serialize=False)),
+                ('channel', models.CharField(max_length=100)),
+                ('version', models.CharField(max_length=100)),
+                ('ping_type', models.CharField(max_length=100)),
+                ('os', models.CharField(max_length=100)),
+                ('build_id', models.CharField(max_length=100)),
+                ('build_date', models.DateTimeField(null=True)),
+                ('metric', models.CharField(max_length=200)),
+                ('metric_type', models.CharField(max_length=100)),
+                ('metric_key', models.CharField(blank=True, max_length=200)),
+                ('client_agg_type', models.CharField(blank=True, max_length=100)),
+                ('total_users', models.IntegerField()),
+                ('histogram', models.TextField(blank=True, null=True)),
+                ('percentiles', models.TextField(blank=True, null=True)),
+            ],
+            options={
+                'db_table': 'view_glam_fog_aggregation',
+                'managed': False,
+            },
+        ),
+        
+    ]
```

Comment:
I'm not sure if you edited these by hand or not? But this migration shouldn't exist. We used to have 2 rows, one with the `histogram` and one with the `percentiles` and then we combined them. This migration was for the existing tables at the time. Since the new models already have this change, these migrations aren't needed and you should run `makemigrations` from the models.  See https://docs.djangoproject.com/en/4.0/topics/migrations/

Reply:
ok. Makes sense. I will re-generate the migrations.